discussions-server/src/main/java/org/xwiki/contrib/discussions/internal/server/DefaultDiscussionMessageRequestCreator.java

```diff
@@ -0,0 +1,184 @@
+import org.xwiki.rendering.parser.ParseException;
+import org.xwiki.rendering.syntax.Syntax;
+import org.xwiki.wysiwyg.converter.HTMLConverter;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.web.EditForm;
+
+import static org.xwiki.rendering.syntax.Syntax.XWIKI_2_0;
+
+/**
+ * Default implementation of {@link DiscussionMessageRequestCreator}.
+ *
+ * @version $Id$
+ * @since 2.1
+ */
+@Component
+@Singleton
+public class DefaultDiscussionMessageRequestCreator implements DiscussionMessageRequestCreator
+{
+    @Inject
+    private DiscussionService discussionService;
+
+    @Inject
+    private DiscussionReferencesResolver discussionReferencesResolver;
+
+    @Inject
+    private HTMLConverter htmlConverter;
+
+    @Inject
+    private MessageService messageService;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private EntityReferenceSerializer<String> entityReferenceSerializer;
+
+    @Override
+    public Optional<Message> createMessage(HttpServletRequest request) throws DiscussionServerException
+    {
+        String serializedReference = request.getParameter(DISCUSSION_REFERENCE_PARAM);
+        if (StringUtils.isBlank(serializedReference)) {
+            throw new DiscussionServerException(HttpServletResponse.SC_BAD_REQUEST,
+                "The discussion reference has not been provided.");
+        } else {
+            DiscussionReference discussionReference =
+                this.discussionReferencesResolver.resolve(serializedReference, DiscussionReference.class);
+            Optional<Discussion> discussionOptional = this.discussionService.get(discussionReference);
+            if (!discussionOptional.isPresent()) {
+                throw new DiscussionServerException(HttpServletResponse.SC_NOT_FOUND,
+                    String.format("Cannot find discussion with reference [%s]", serializedReference));
+            } else {
+                return this.createMessage(discussionOptional.get(), request);
+            }
+        }
+    }
+
+    private Optional<Message> createMessage(Discussion discussion, HttpServletRequest request)
+    {
+        String content = getContent(request);
+        Syntax syntax = getSyntax(request);
+        EditForm editForm = this.prepareForm(request);
+        DiscussionStoreConfigurationParameters parameters = new DiscussionStoreConfigurationParameters();
+        request.getParameterMap().forEach((key, value) -> {
+            if (key.startsWith(STORE_CONFIGURATION_PARAMETER_PREFIX)) {
+                String parameterKey = key.substring(STORE_CONFIGURATION_PARAMETER_PREFIX.length());
+                parameters.put(parameterKey, value);
+            }
+        });
+
+        // Handle temporary uploads
+        List<String> temporaryUploadedFiles = editForm.getTemporaryUploadedFiles();
+        parameters.put(TEMPORARY_UPLOADED_ATTACHMENTS, temporaryUploadedFiles);
+        Optional<Message> messageOptional;
```

Comment:
Feels a bit strange to declare that here and assign it at the end of the method.

Reply:
indeed, probably historical reasons  since this method has evolved a lot. I'll move it